timed/authentication.py

```diff
@@ -29,42 +26,39 @@ def get_introspection(self, access_token, id_token, payload):
             verify=settings.OIDC_VERIFY_SSL,
             headers=headers,
             data={"token": access_token},
+            timeout=10,
         )
         response.raise_for_status()
         return response.json()
 
     def get_userinfo_or_introspection(self, access_token):
         try:
-            claims = self.cached_request(
-                self.get_userinfo, access_token, "auth.userinfo"
-            )
-            return claims
-        except requests.HTTPError as e:
-            if e.response.status_code not in [401, 403]:
-                raise e
+            return self.cached_request(self.get_userinfo, access_token, "auth.userinfo")
+        except requests.HTTPError as exc:
+            if exc.response.status_code not in [401, 403]:
+                raise
             if settings.OIDC_CHECK_INTROSPECT:
                 try:
                     # check introspection if userinfo fails (confidential client)
                     claims = self.cached_request(
                         self.get_introspection, access_token, "auth.introspection"
                     )
                     if "client_id" not in claims:
-                        raise SuspiciousOperation(
-                            "client_id not present in introspection"
-                        )
-                    return claims
+                        msg = "client_id not present in introspection"
+                        raise SuspiciousOperation(msg)
                 except requests.HTTPError as e:
                     # if the authorization fails it's not a valid client or
                     # the token is expired and permission is denied.
                     # Handing on the 401 Client Error would be transformed into
                     # a 500 by Django's exception handling. But that's not what we want.
                     if e.response.status_code not in [401, 403]:  # pragma: no cover
-                        raise e
-            raise AuthenticationFailed()
+                        raise
+                else:
+                    return claims
+            raise AuthenticationFailed from exc
 
-    def get_or_create_user(self, access_token, id_token, payload):
+    def get_or_create_user(self, access_token, *args, **kwargs):
```

Comment:
I've thought a good bit about this change (and similar ones elsewhere). I'm now of the opinion that we should keep the old named params, as they're part of the *interface*, even if not used in this particular implementation. Instead either disable the offending rule or do a #noqa marker. 

`*args, **kwargs` is for when you don't know what you might get as parameters, and possibly passing it on, but we *do* know the parameters, we just don't use them. Therefore IMNSHO they should be kept as before

Reply:
i changed them back, respectively used `_{argname}` instead of `{argname}` where possible

https://github.com/adfinis/timed-backend/pull/1049/commits/717820e0c781b888aafa76ff0605e50ab7f92378